### ISSUE TITLE
User guide and keyboard shortcuts documentation

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5420]
In the Classic editor, there's the Help section with a (limited) user guide about:
- editing the title and the post
- inserting media
- publish settings
- discussion settings

![screen shot 2018-03-05 at 21 08 50](https://user-images.githubusercontent.com/1682452/36997438-e7e1e046-20b9-11e8-9d81-9f0b02680f53.png)

there's also a modal with a list of all the available keyboard shortcuts:

![screen shot 2018-03-05 at 21 09 12](https://user-images.githubusercontent.com/1682452/36997439-e80320e4-20b9-11e8-8326-5f496566549a.png)

To my knowledge, in Gutenberg there's no help or user guide at all, not sure if this was discussed at some point. I'd strongly recommend to start thinking at a user guide. Whether it lives in the admin or on an external page / handbook, it should be easily discoverable.

As per keyboard and screen reader users, they need a well documented list of all the available keyboard shortcuts in Gutenberg, plus a guide to some of the available accessibility features, e.g. the Ctrl + backtick shortcut to navigate through the main editor regions.

We've discussed this during today's accessibility meeting and agreed the list of keyboard shortcuts is essential to effectively use the editor and should be available since version 1.

**Comment by afercia** [comment]
Related:
https://github.com/WordPress/gutenberg/issues/71 and https://github.com/WordPress/gutenberg/issues/2980
not the same thing though, since 2980 is about displaying shortcuts within the various menus, close to the related control.

**Comment by afercia** [comment]
See also https://github.com/WordPress/gutenberg/pull/4411 for some more shortcuts.

**Comment by nic-bertino** [comment]
Building on #3218, I think this UI element should incorporate customization as well (per https://github.com/WordPress/gutenberg/issues/3218#issuecomment-369668603), as it keeps that control close to the user.

Choosing where to fold this into the UI might be a bit of a challenge, but for now, it might make the most sense to have it under the post triple dot:

![image](https://user-images.githubusercontent.com/883093/37048470-d6f428a2-2122-11e8-8b06-c420f6065bba.png)

Again, I certainly think there are better ways to introduce the keyboard shortcut palette, and I'll continue mocking up some suggestions.

**Comment by postphotos** [comment]
@nic-bertino , maybe it can be under Tools > **Keyboard Shortcuts**? It makes sense to have a similar modal, a la Classic Editor.

**Comment by afercia** [comment]
Considering a list of keyboard shortcuts is mainly meant for keyboard and assistive technologies users, I'd consider to improve the current experience and avoid a modal. Modals are always tricky for those users.

Adding the design label because both the help / user guide and the list of shortcuts need some thinking 🙂

**Comment by karmatosed** [comment]
@afercia with regards to the design of this, are there any accessible good examples in other applications we could learn from?

**Comment by afercia** [comment]
@karmatosed not off the top of my head. In Gmail for example, typing `?` makes a list of keyboard shortcuts appear but that's not exactly what I'd like to see in Gutenberg. Maybe we should first consider if we want the Help section in Gutenbreg or on .org.
In the first case, maybe use the "screen takeover" (the one meant for plugins)? Regardless, I'd tend to think the most important thing is that the link or button to access the Help section should be easily discoverable. /Cc @rianrietveld for her experience in documentation 🙂

**Comment by rianrietveld** [comment]
My experience is that not many people discover the help text, as it is now in the classic editor.
So a better icon/text to help text is definitely needed.

Thinking of all the stuff we need to explain, I wonder if we should integrate all the help text inside the Gutenberg editor. That will become a usability nightmare I guess, fitting everything in modals, screen takeover, whatever. 

Thinking out load and maybe totally out of context:
All the shortcuts, AT manuals, etc can be placed in documentation on WordPress.org. What not just have a clear icon for help and link to the docs applicable in the right language? So keep everything on .org (and translate there too) . And don't burden the UI with complicated constructions.

**Comment by karmatosed** [comment]
> My experience is that not many people discover the help text, as it is now in the classic editor.
So a better icon/text to help text is definitely needed.

This is pretty reflective of my own experience.

**Comment by rianrietveld** [comment]
@karmatosed 
> do you imagine it as a link to an external source?
Yes, make it as simple as that

**Comment by mtias** [comment]
We now have both the modal listing available shortcuts and "tips" guiding specific areas of the interface. Let's focus any ideas and enhancements on new issues.
